### PR TITLE
Important Routes Added

### PR DESCRIPTION
**Pull request by RiddhiGupta5**

### CODE REVIEW

File: app/admin_views.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                # Creating superuser (Admin)
+                User.objects.create_superuser(
+                    email=req_data['email'],
+                    username=req_data['username'],
+                    password=req_data['password'],
+                    insta_handle=None,
+                    platform=0 
+                )
+                user = User.objects.get(email=req_data['email'])
+                return Response({"message":"Admin Signed up"}, status=status.HTTP_400_BAD_REQUEST)
```
Comment (L04DB4L4NC3R) [inline app/admin_views.py]: Why is 400 being returned here?

File: app/admin_views.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        else:            
+            return Response({"message":"Invalid Email and Password"}, status=status.HTTP_400_BAD_REQUEST)
+
+# View to Login Admin
+class AdminLoginView(APIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        req_data = request.data
```
Comment (L04DB4L4NC3R) [inline app/admin_views.py]: Why is this line required? You can directly get the values from `request.data`

File: app/admin_views.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+from .models import User, Question, Answer
+from .serializers import QuestionSerializer
+
+# View for Admin Signup
+class AdminSignupView(APIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        req_data = {}
```
Comment (L04DB4L4NC3R) [inline app/admin_views.py]: Why is this req_data being used. `request.data` can be used without reassignment

File: app/admin_views.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            else:
+                return Response({"message":"Not an Admin"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+# View to Logout Admin
+class AdminLogoutView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, format=None):
+        user = request.user
```
Comment (L04DB4L4NC3R) [inline app/admin_views.py]: `user` not being used anywhere

File: app/admin_views.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        response = {"message":"Admin logged out"}
+        request.user.auth_token.delete()
+        return Response(response, status=status.HTTP_200_OK)
+
+class QuestionView(APIView):
+
+    def post(self, request):
+        user = request.user
```
Comment (L04DB4L4NC3R) [inline app/admin_views.py]: Why is this additional assignment needed?

File: app/urls.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+from rest_framework import routers
+
+router = routers.DefaultRouter()
+
+
+urlpatterns = [
+    path("login/", GoogleSignInView.as_view()),
+    path("logout/", LogoutView.as_view()),
+    path("update_insta_handle/", UpdateInstaHandle.as_view()),
+    path("daily_question/", DailyQAView.as_view()),
```
Comment (L04DB4L4NC3R) [inline app/urls.py]: Dont use underscore in the URL, it is bad practice. Instead use a base url. For example `/admin/signup` instead of `/admin_signup`

File: app/views.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authtoken.models import Token
+from rest_framework.permissions import AllowAny, IsAuthenticated
+
+from .models import User, Question, Answer
+
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (L04DB4L4NC3R) [inline app/views.py]: Why is this in views and not the entrypoint?

File: app/views.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+                        continue
+                    else:
+                        user_answer.delete()
+                except Answer.DoesNotExist:
+                    pass 
+                insta_answer.user_id = user
+                insta_answer.save()
+
+            insta_user.delete()
+            for answer in answer_to_delete:
```
Comment (L04DB4L4NC3R) [inline app/views.py:92]: Keep an is_deleted boolean instead of removing everything from the DB

File: challenges_app/settings.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+For more information on this file, see
+https://docs.djangoproject.com/en/3.0/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/3.0/ref/settings/
+"""
+
+import os
+from dotenv import load_dotenv
+load_dotenv()
```
Comment (L04DB4L4NC3R) [inline challenges_app/settings.py:16]: This is correct. `load_dotenv` shouldnt be in the views though

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

**Comment by RiddhiGupta5** [comment]
Made all the minor fixes

File: app/views.py (hunk 8 above)
Comment (RiddhiGupta5) [reply app/views.py:92]: I think that we can actually delete this record because all the details that this insta_user account will have will be transferred to the other account (including the insta handle details). Thus this record or user will simply become redundant data.

Comment (L04DB4L4NC3R) [reply app/views.py:92]: Makes sense

**Review by L04DB4L4NC3R — APPROVED** [review]
